src/components/stimuli/Stimuli.tsx

```diff
@@ -29,15 +44,80 @@ const Stimuli: React.FC = () => {
       }
     }
 
-    fetchStimuliUrls();
-  }, []);
+    if (!isUploading) fetchStimuliUrls();
+  }, [isUploading]);
+
+  const onFileSelect = (e: any) => {
+    const file: File = e.target.files[0];
+    const formData = new FormData();
+    formData.append('file', file);
+    setUploading(true);
+    fetch(`${process.env.REACT_APP_BACKEND_ADDRESS}/image`, {
+      method: 'POST',
+      body: formData,
+    })
+      .then(res => res.json())
+      .then(() => {
+        setUploading(false);
+        setShowToast(true);
+      });
+  };
+
+  const fileSelector: HTMLInputElement = buildFileSelector();
+  fileSelector.onchange = onFileSelect;
+
+  const handleUploadClick = (e: any) => {
+    e.preventDefault();
+    fileSelector.click();
+  };
 
   return (
     <Container>
+      <Row>
+        <Col>
+          <Button
+            variant="secondary"
+            size="lg"
+            className="my-3"
+            onClick={handleUploadClick}
+            block
+          >
+            {isUploading ? (
+              <Spinner
+                as="span"
+                animation="border"
+                role="status"
+                aria-hidden="true"
+              />
+            ) : (
+              'Upload New Image'
+            )}
+          </Button>
```

Comment:
Rather than using a button component, we might consider using an input group. These components are built for handling user input so it may simplify things a bit. Something like the [default file input shown here](https://mdbootstrap.com/docs/react/forms/file-input/) might work better. This way there's a confirmation that the user selected the right file before they click upload. This is a more subjective design decision so we can discuss as a group if you'd like.